packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -57,37 +54,21 @@ export const createPluginActions: CreatePluginActions = (
   //  Because the container doesn't send the full state in its messages, we need to track it ourselves.
   //  isModal and height is not included in the messages to the children and must thus be tracked here.
   //  In future improved versions of the plugin API, this should not be needed.
-  let state: FieldPluginData = defaultState
 
-  let assetSelectedCallbackRef: undefined | ((filename: Asset) => void) =
-    undefined
-  let assetSelectedCallbackId: undefined | string = undefined
+  const { pushCallback, popCallback } = callbackQueue()
 
   const onStateChange: OnStateChangeMessage = (data) => {
-    state = {
-      ...state,
-      ...partialPluginStateFromStateChangeMessage(data),
-    }
-    onUpdateState(state)
+    popCallback('state', data.callbackId)?.(data)
+    onUpdateState(pluginStateFromStateChangeMessage(data))
+  }
+  const onLoaded: OnLoadedMessage = (data) => {
+    onUpdateState(pluginStateFromStateChangeMessage(data))
   }
```

Comment:
This is a new message. Whenever the state in the field plugin editor changes, it will send an event with the new state to the field plugin. The state is identical to the state in the `loaded` message.